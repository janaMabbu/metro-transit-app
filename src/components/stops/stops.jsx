import React, { PureComponent, Fragment } from 'react'
import { connect } from 'react-redux'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { getStops, isGetStopsSuccessful } from 'ducks/get-stops'

export class Stops extends PureComponent {
  static propTypes = {
    stops: ImmutablePropTypes.list,
    selectedStop:PropTypes.string,
    setSelectedDirection: PropTypes.func.isRequired,
    isGetStopsSuccessful: PropTypes.bool
  }


  setSelectedStop = (event) => {
    const { setSelectedStop } = this.props
    setSelectedStop(event.target.value)
  }


  render () {
    const { stops } = this.props
    return (
      <Fragment>
       <ul>
        { stops.map(stop => <li key={stop.get('PlaceCode')} value={stop.get('PlaceCode')} >{`${stop.get('PlaceCode')} -- ${stop.get('Description')}`}</li>) }
       </ul>
      </Fragment>
    )
  }

  render () {
    return (
      <Fragment>
       {this.props.isGetStopsSuccessful ? this.renderContent(): this.renderError()}
      </Fragment>
    )
  }

  renderContent = () => {
    const { stops } = this.props
    return (
      <Fragment>
       <ul>
        { stops.map(stop => <li key={stop.get('PlaceCode')} value={stop.get('PlaceCode')} >{`${stop.get('PlaceCode')} -- ${stop.get('Description')}`}</li>) }
       </ul>
      </Fragment>
    )
  }

  renderError = () => {
     return(
      <Fragment>
        something went wrong loading the directions for the stops, please reselct the direction and route from the dropdown!!!
      </Fragment>
    )
  }

}
const mapStateToProps = state => {
  return {
    stops: getStops(state),
    isGetStopsSuccessful: isGetStopsSuccessful(state)
  }
}
const mapDispatchToProps = {}
export default connect(mapStateToProps, mapDispatchToProps)(Stops)
