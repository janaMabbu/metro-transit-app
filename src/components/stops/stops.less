@import "~less/common.less";

.stops-container {
    background-color: #f5f5f4;
    padding: 20px;
    color: #626462;
    text-align: left;
    width: 100%;
    margin: 0 auto;
    margin-top: 20px;
    font-size: 16px;

  &__title {
    font-size: 34px;
    margin-bottom: 15px;
    line-height: 1.2;
    color: #626462;
    text-align: left;
    padding: 0;
    margin: 0 0 10px;
  }

  &__list {
    margin: 30px 0px;
  }

  &__stop-description {
    padding: 12px 0 12px;
    border-bottom: 1px solid #8a8b8a;
  }
  &__stop-info {
    font-weight: 700;
    line-height: 1.2;
    font-size: 32px;
    margin: 0 0 48px;
    color: @light-grey;
  }

  &__button {
     color: #626462;
     line-height: 1.5;
     font-weight: 700;
     font-size: 16px;
     border:none;
     background: none;
     margin-top: 20px;
     padding: 0;
  }

  &__expand, &__closed {
    vertical-align: middle;
    width: 20px;
    height: 20px;
    background: url('../../images/circle-plus-blue.svg') no-repeat center;
    background-size: contain;
    margin-right: 10px;
    display: inline-block;
  }

    &__expand {
      background: url('../../images/circle-minus-blue.svg') no-repeat center;
      background-size: contain;
    }

}
