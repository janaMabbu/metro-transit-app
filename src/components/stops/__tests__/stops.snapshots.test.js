import { shallow, mount } from 'enzyme'
import React from 'react'
import Immutable from 'immutable'

import { Stops } from 'components/stops/stops'

describe('Component - Stops', () => {
  let props
  const stops = [
    {
      'PlaceCode': 'PlaceCode1',
      'Description': 'Description'
    },
     {
      'PlaceCode': 'PlaceCode2',
      'Description': 'Description11'
    },
     {
      'PlaceCode': 'PlaceCode3',
      'Description': 'Description22'
    },
      {
      'PlaceCode': 'PlaceCode4',
      'Description': 'Description'
    },
     {
      'PlaceCode': 'PlaceCode5',
      'Description': 'Description11'
    },
     {
      'PlaceCode': 'PlaceCode6',
      'Description': 'Description22'
    },
  ]
  beforeEach(() => {
    props = {
      stops: Immutable.fromJS(stops),
      selectedRoute:'xyz',
      selectedRoute:'abc',
      isGetStopsSuccessful: true
    }
  })
  it('should render the Stops snapshot', () => {
    const wrapper = shallow(<Stops {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the Stops fallback', () => {
    const wrapper = shallow(<Stops {...props} />)
    wrapper.setProps({ isGetStopsSuccessful: false })
    expect(wrapper.contains('something went wrong loading the directions for the stops, please reselct the direction and route from the dropdown!!!')).toBe(true)
  })

  it('should only display first four stops', () => {
    const wrapper = shallow(<Stops {...props}/>)
    expect(wrapper.find('.stops-container__stop-description').length).toBe(4)
  })

  it('should only all first 6 stops on click on showmore and 4 on clicck of showless', () => {
    const wrapper = shallow(<Stops {...props}/>)
    wrapper.find('.stops-container__button').first().simulate('click')
    expect(wrapper.find('.stops-container__stop-description').length).toBe(6)
    wrapper.find('.stops-container__button').first().simulate('click')
    expect(wrapper.find('.stops-container__stop-description').length).toBe(4)
  })

})