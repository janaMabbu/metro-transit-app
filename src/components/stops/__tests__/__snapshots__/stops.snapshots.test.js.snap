// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Stops should render the Stops snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stops
    isGetStopsSuccessful={true}
    selectedRoute="abc"
    stops={
      Immutable.List [
        Immutable.Map {
          "PlaceCode": "PlaceCode1",
          "Description": "Description",
        },
        Immutable.Map {
          "PlaceCode": "PlaceCode2",
          "Description": "Description11",
        },
        Immutable.Map {
          "PlaceCode": "PlaceCode3",
          "Description": "Description22",
        },
        Immutable.Map {
          "PlaceCode": "PlaceCode4",
          "Description": "Description",
        },
        Immutable.Map {
          "PlaceCode": "PlaceCode5",
          "Description": "Description11",
        },
        Immutable.Map {
          "PlaceCode": "PlaceCode6",
          "Description": "Description22",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="stops-container"
      >
        <div
          className="stops-container__info"
        >
          <h2
            className="stops-container__title"
          >
            Stops
          </h2>
          <div>
            RouteId: 
            abc
          </div>
          <div>
            DirectionId: 
          </div>
        </div>
        Immutable.List [
          <div
            className="stops-container__list"
            role="list"
          >
            <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode1
                   
                </strong>
              </span>
              <span>
                Description
              </span>
            </div>
          </div>,
          <div
            className="stops-container__list"
            role="list"
          >
            <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode2
                   
                </strong>
              </span>
              <span>
                Description11
              </span>
            </div>
          </div>,
          <div
            className="stops-container__list"
            role="list"
          >
            <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode3
                   
                </strong>
              </span>
              <span>
                Description22
              </span>
            </div>
          </div>,
          <div
            className="stops-container__list"
            role="list"
          >
            <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode4
                   
                </strong>
              </span>
              <span>
                Description
              </span>
            </div>
          </div>,
        ]
        <button
          className="stops-container__button"
          onClick={[Function]}
        >
          <span
            className="stops-container__closed"
          />
          Show 
          more
           Stops
        </button>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stops-container__info"
          >
            <h2
              className="stops-container__title"
            >
              Stops
            </h2>
            <div>
              RouteId: 
              abc
            </div>
            <div>
              DirectionId: 
            </div>
          </div>,
          Immutable.List [
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode1
                     
                  </strong>
                </span>
                <span>
                  Description
                </span>
              </div>
            </div>,
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode2
                     
                  </strong>
                </span>
                <span>
                  Description11
                </span>
              </div>
            </div>,
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode3
                     
                  </strong>
                </span>
                <span>
                  Description22
                </span>
              </div>
            </div>,
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode4
                     
                  </strong>
                </span>
                <span>
                  Description
                </span>
              </div>
            </div>,
          ],
          <button
            className="stops-container__button"
            onClick={[Function]}
          >
            <span
              className="stops-container__closed"
            />
            Show 
            more
             Stops
          </button>,
        ],
        "className": "stops-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="stops-container__title"
              >
                Stops
              </h2>,
              <div>
                RouteId: 
                abc
              </div>,
              <div>
                DirectionId: 
              </div>,
            ],
            "className": "stops-container__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Stops",
                "className": "stops-container__title",
              },
              "ref": null,
              "rendered": "Stops",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "RouteId: ",
                  "abc",
                ],
              },
              "ref": null,
              "rendered": Array [
                "RouteId: ",
                "abc",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "DirectionId: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "DirectionId: ",
                undefined,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "PlaceCode1",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode1
                   
                </strong>
              </span>
              <span>
                Description
              </span>
            </div>,
            "className": "stops-container__list",
            "role": "list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  <strong>
                    PlaceCode1
                     
                  </strong>
                </span>,
                <span>
                  Description
                </span>,
              ],
              "className": "stops-container__stop-description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    PlaceCode1
                     
                  </strong>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "PlaceCode1",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "PlaceCode1",
                    " ",
                  ],
                  "type": "strong",
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description",
                },
                "ref": null,
                "rendered": "Description",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "PlaceCode2",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode2
                   
                </strong>
              </span>
              <span>
                Description11
              </span>
            </div>,
            "className": "stops-container__list",
            "role": "list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  <strong>
                    PlaceCode2
                     
                  </strong>
                </span>,
                <span>
                  Description11
                </span>,
              ],
              "className": "stops-container__stop-description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    PlaceCode2
                     
                  </strong>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "PlaceCode2",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "PlaceCode2",
                    " ",
                  ],
                  "type": "strong",
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description11",
                },
                "ref": null,
                "rendered": "Description11",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "PlaceCode3",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode3
                   
                </strong>
              </span>
              <span>
                Description22
              </span>
            </div>,
            "className": "stops-container__list",
            "role": "list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  <strong>
                    PlaceCode3
                     
                  </strong>
                </span>,
                <span>
                  Description22
                </span>,
              ],
              "className": "stops-container__stop-description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    PlaceCode3
                     
                  </strong>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "PlaceCode3",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "PlaceCode3",
                    " ",
                  ],
                  "type": "strong",
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description22",
                },
                "ref": null,
                "rendered": "Description22",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "PlaceCode4",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="stops-container__stop-description"
            >
              <span>
                <strong>
                  PlaceCode4
                   
                </strong>
              </span>
              <span>
                Description
              </span>
            </div>,
            "className": "stops-container__list",
            "role": "list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  <strong>
                    PlaceCode4
                     
                  </strong>
                </span>,
                <span>
                  Description
                </span>,
              ],
              "className": "stops-container__stop-description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    PlaceCode4
                     
                  </strong>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "PlaceCode4",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "PlaceCode4",
                    " ",
                  ],
                  "type": "strong",
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description",
                },
                "ref": null,
                "rendered": "Description",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="stops-container__closed"
              />,
              "Show ",
              "more",
              " Stops",
            ],
            "className": "stops-container__button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "stops-container__closed",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "Show ",
            "more",
            " Stops",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="stops-container"
        >
          <div
            className="stops-container__info"
          >
            <h2
              className="stops-container__title"
            >
              Stops
            </h2>
            <div>
              RouteId: 
              abc
            </div>
            <div>
              DirectionId: 
            </div>
          </div>
          Immutable.List [
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode1
                     
                  </strong>
                </span>
                <span>
                  Description
                </span>
              </div>
            </div>,
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode2
                     
                  </strong>
                </span>
                <span>
                  Description11
                </span>
              </div>
            </div>,
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode3
                     
                  </strong>
                </span>
                <span>
                  Description22
                </span>
              </div>
            </div>,
            <div
              className="stops-container__list"
              role="list"
            >
              <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode4
                     
                  </strong>
                </span>
                <span>
                  Description
                </span>
              </div>
            </div>,
          ]
          <button
            className="stops-container__button"
            onClick={[Function]}
          >
            <span
              className="stops-container__closed"
            />
            Show 
            more
             Stops
          </button>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="stops-container__info"
            >
              <h2
                className="stops-container__title"
              >
                Stops
              </h2>
              <div>
                RouteId: 
                abc
              </div>
              <div>
                DirectionId: 
              </div>
            </div>,
            Immutable.List [
              <div
                className="stops-container__list"
                role="list"
              >
                <div
                  className="stops-container__stop-description"
                >
                  <span>
                    <strong>
                      PlaceCode1
                       
                    </strong>
                  </span>
                  <span>
                    Description
                  </span>
                </div>
              </div>,
              <div
                className="stops-container__list"
                role="list"
              >
                <div
                  className="stops-container__stop-description"
                >
                  <span>
                    <strong>
                      PlaceCode2
                       
                    </strong>
                  </span>
                  <span>
                    Description11
                  </span>
                </div>
              </div>,
              <div
                className="stops-container__list"
                role="list"
              >
                <div
                  className="stops-container__stop-description"
                >
                  <span>
                    <strong>
                      PlaceCode3
                       
                    </strong>
                  </span>
                  <span>
                    Description22
                  </span>
                </div>
              </div>,
              <div
                className="stops-container__list"
                role="list"
              >
                <div
                  className="stops-container__stop-description"
                >
                  <span>
                    <strong>
                      PlaceCode4
                       
                    </strong>
                  </span>
                  <span>
                    Description
                  </span>
                </div>
              </div>,
            ],
            <button
              className="stops-container__button"
              onClick={[Function]}
            >
              <span
                className="stops-container__closed"
              />
              Show 
              more
               Stops
            </button>,
          ],
          "className": "stops-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="stops-container__title"
                >
                  Stops
                </h2>,
                <div>
                  RouteId: 
                  abc
                </div>,
                <div>
                  DirectionId: 
                </div>,
              ],
              "className": "stops-container__info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Stops",
                  "className": "stops-container__title",
                },
                "ref": null,
                "rendered": "Stops",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "RouteId: ",
                    "abc",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "RouteId: ",
                  "abc",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "DirectionId: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "DirectionId: ",
                  undefined,
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "PlaceCode1",
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode1
                     
                  </strong>
                </span>
                <span>
                  Description
                </span>
              </div>,
              "className": "stops-container__list",
              "role": "list",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    <strong>
                      PlaceCode1
                       
                    </strong>
                  </span>,
                  <span>
                    Description
                  </span>,
                ],
                "className": "stops-container__stop-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <strong>
                      PlaceCode1
                       
                    </strong>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "PlaceCode1",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "PlaceCode1",
                      " ",
                    ],
                    "type": "strong",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "PlaceCode2",
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode2
                     
                  </strong>
                </span>
                <span>
                  Description11
                </span>
              </div>,
              "className": "stops-container__list",
              "role": "list",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    <strong>
                      PlaceCode2
                       
                    </strong>
                  </span>,
                  <span>
                    Description11
                  </span>,
                ],
                "className": "stops-container__stop-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <strong>
                      PlaceCode2
                       
                    </strong>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "PlaceCode2",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "PlaceCode2",
                      " ",
                    ],
                    "type": "strong",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description11",
                  },
                  "ref": null,
                  "rendered": "Description11",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "PlaceCode3",
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode3
                     
                  </strong>
                </span>
                <span>
                  Description22
                </span>
              </div>,
              "className": "stops-container__list",
              "role": "list",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    <strong>
                      PlaceCode3
                       
                    </strong>
                  </span>,
                  <span>
                    Description22
                  </span>,
                ],
                "className": "stops-container__stop-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <strong>
                      PlaceCode3
                       
                    </strong>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "PlaceCode3",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "PlaceCode3",
                      " ",
                    ],
                    "type": "strong",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description22",
                  },
                  "ref": null,
                  "rendered": "Description22",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "PlaceCode4",
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="stops-container__stop-description"
              >
                <span>
                  <strong>
                    PlaceCode4
                     
                  </strong>
                </span>
                <span>
                  Description
                </span>
              </div>,
              "className": "stops-container__list",
              "role": "list",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    <strong>
                      PlaceCode4
                       
                    </strong>
                  </span>,
                  <span>
                    Description
                  </span>,
                ],
                "className": "stops-container__stop-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <strong>
                      PlaceCode4
                       
                    </strong>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "PlaceCode4",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "PlaceCode4",
                      " ",
                    ],
                    "type": "strong",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="stops-container__closed"
                />,
                "Show ",
                "more",
                " Stops",
              ],
              "className": "stops-container__button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "stops-container__closed",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "Show ",
              "more",
              " Stops",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
